data/cube/aggregate/AverageStrictAggregator.js

```diff
@@ -10,15 +10,21 @@ import {Aggregator} from '@xh/hoist/data/cube/aggregate/Aggregator';
 export class AverageStrictAggregator extends Aggregator {
 
     aggregate(rows, fieldName) {
-        let total = null;
+        let strict = true,
+            total = null,
+            count = 0;
 
-        for (const row of rows) {
+        this.forEachLeaf(rows, row => {
             const val = row.data[fieldName];
-            if (val == null) return null;
+            if (val == null) {
+                strict = false;
+                return;
```

Comment:
Returning null from here was not exiting this  aggregate function leading to line 27 executing incorrectly.